apps/rush-lib/src/utilities/Utilities.ts

```diff
@@ -120,6 +121,26 @@ export class Utilities {
     return homeFolder;
   }
 
+  /**
+   * Returns true if rush is running on a CI.
+    *
+   * The package is-ci is to detect this.
+   * Set environment variable RUSH_CI_MODE to override:
+   *   To force CI mode, set the environment variable to CI
+   *   To force non-CI mode, set the environment variable to NON-CI
+   */
+  public static isCI(): boolean {
```

Comment:
I think this should return an `enum` with more information about the CI mode. Something like

```typescript
enum RushCIMode {
  nonCI,
  detectedCI,
  overrideDisabledCI,
  overrideEnabledCI
}
```

Reply:
I think implementing this will complicate the code flow, especially if developers choose to implement different behaviors for detectedCI vs overrideEnabledCI. That would make control flow tracing and debugging much harder.